core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -1265,6 +1292,21 @@ void addPath(String path, FolderDetails details) {
     public void setOptions(Map<String, String> options) {
       this.options = options;
     }
+
+    public List<SegmentMinMax> getSegmentMinMax() {
+      List<SegmentMinMax> segmentMinMaxList = null;
+      try {
+        segmentMinMaxList =
+            (List<SegmentMinMax>) ObjectSerializationUtil.convertStringToObject(segmentMinMax);
+      } catch (IOException e) {
+        LOGGER.error("Error while getting segment minmax");
+      }
+      return segmentMinMaxList;
```

Comment:
i recommend to return empty list instead of null and all the place remove null check

Reply:
done